justfile

```diff
@@ -1,55 +1,100 @@
+# just has no idiom for setting an default value for an environment variable
+# so we shell out, as we need VIRTUAL_ENV in the justfile environment
+export VIRTUAL_ENV  := `echo ${VIRTUAL_ENV:-.venv}`
+
+# TODO: make it /scripts on windows?
+export BIN := VIRTUAL_ENV + "/bin"
+export PIP := BIN + "/python -m pip"
+# enforce our chosed pip compile flags
+export COMPILE := BIN + "/pip-compile --allow-unsafe --generate-hashes"
+
+
 # list available commands
 default:
-    @just --list
+    @{{ just_executable() }} --list
+
+
+# clean up temporary files
+clean:
+    rm -rf .venv
 
-# Set up local dev environment
-dev_setup:
+
+# ensure valid virtualenv
+virtualenv:
     #!/usr/bin/env bash
-    set -euo pipefail
-    . scripts/setup_functions
-    dev_setup
+    # allow users to specify python version in .env
+    PYTHON_VERSION=${PYTHON_VERSION:-python3.9}
+    # create venv and upgrade pip
+    test -d $VIRTUAL_ENV || { $PYTHON_VERSION -m venv $VIRTUAL_ENV && $PIP install --upgrade pip; }
+    # ensure we have pip-tools so we can run pip-compile
+    test -e $BIN/pip-compile || $PIP install pip-tools
```

Comment:
If it doesn't break just could we put a blank line before each comment line?  Really helps with legibility if you find blocks merge together 😃

Reply:
will try it and see

Reply:
it works, updated